src/AggregateFunctions/Store.h

```diff
@@ -0,0 +1,267 @@
+        if (count != 0)
+        {
+            num_bins = max_key - min_key + 1;
+        }
+
+        size_t sparse_encoding_overhead = 0;
+        for (int index = min_key; index <= max_key; ++index)
+        {
+            if (bins[index - offset] != 0)
+            {
+                num_non_empty_bins++;
+                sparse_encoding_overhead += 2; // 2 bytes for index delta
+            }
+        }
+
+        size_t dense_encoding_overhead = (num_bins - num_non_empty_bins) * estimatedFloatSize(0.0);
+
+        // Choose the smallest encoding and write to buffer
+        if (dense_encoding_overhead <= sparse_encoding_overhead)
+        {
+            // Write the dense encoding
+            writeBinary(enc.BinEncodingContiguousCounts, buf); // Flag for dense encoding
+            writeVarUInt(num_bins, buf);
+            writeVarInt(min_key, buf);
+            writeVarInt(1, buf); // indexDelta in dense encoding
+            for (int index = min_key; index <= max_key; ++index)
+            {
+                writeFloatBinary(bins[index - offset], buf);
+            }
+        }
+        else
+        {
+            // Write the sparse encoding
+            writeBinary(enc.BinEncodingIndexDeltasAndCounts, buf); // Flag for sparse encoding
+            writeVarUInt(num_non_empty_bins, buf);
+            int previous_index = 0;
+            for (int index = min_key; index <= max_key; ++index)
+            {
+                Float64 count = bins[index - offset];
+                if (count != 0)
+                {
+                    writeVarInt(index - previous_index, buf);
+                    writeFloatBinary(count, buf);
+                    previous_index = index;
+                }
+            }
+        }
+    }
+
+    void deserialize(ReadBuffer& buf)
+    {
+        UInt8 encoding_mode;
+        readBinary(encoding_mode, buf);
+        if (encoding_mode == enc.BinEncodingContiguousCounts)
+        {
+            UInt64 num_bins;
+            readVarUInt(num_bins, buf);
+            int min_key;
+            readVarInt(min_key, buf);
+            int index_delta;
+            readVarInt(index_delta, buf);
+
+            for (UInt64 i = 0; i < num_bins; ++i)
+            {
+                Float64 count;
+                readFloatBinary(count, buf);
+                add(min_key, count);
+                min_key += index_delta;
+            }
+        }
+        else
+        {
+            UInt64 num_non_empty_bins;
+            readVarUInt(num_non_empty_bins, buf);
+            int previous_index = 0;
+            for (UInt64 i = 0; i < num_non_empty_bins; ++i)
+            {
+                int index_delta;
+                readVarInt(index_delta, buf);
+                Float64 count;
+                readFloatBinary(count, buf);
+                previous_index += index_delta;
+                add(previous_index, count);
+            }
+        }
+    }
+
+private:
+    int chunk_size;
```

Comment:
It's better to avoid using a signed implementation-defined integer type. If you need a type for sizes, UInt64 is ok; if you need a 32-bit integer type for sizes, UInt32 is ok.